### PR TITLE
Make thirdPartyImage optional

### LINKED ISSUE #968: Webhook validation fails with missing required field "thirdPartyImage"
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

**What did you expect to happen:**
```
error: error validating "STDIN": error validating data: ValidationError(AzureMachineTemplate.spec.template.spec.image.marketplace): missing required field "thirdPartyImage" in io.x-k8s.cluster.infrastructure.v1alpha3.AzureMachineTemplate.spec.template.spec.image.marketplace; if you choose to ignore these errors, turn validation off with --validate=false
```

The field should default to false or not be required.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #968

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Make thirdPartyImage optional
```

### CODE REVIEW

File: api/v1alpha3/types.go [hunk 1]
```diff
@@ -261,6 +261,7 @@ type AzureMarketplaceImage struct {
 	// ThirdPartyImage indicates the image is published by a third party publisher and a Plan
 	// will be generated for it.
 	// +kubebuilder:default=false
+	// +optional
 	ThirdPartyImage bool `json:"thirdPartyImage"`
```
Comment (nader-ziada) [inline api/v1alpha3/types.go:265]: does bool need `omitempty`?

Comment (CecileRobertMichon) [reply api/v1alpha3/types.go:265]: it shouldn't, empty bool is false

**Pull request merged**